base/strings/faces.jl

```diff
@@ -0,0 +1,779 @@
+              parse(Int, rgb[4:5], base=16),
+              parse(Int, rgb[6:7], base=16))
+    elseif startswith(rgb, 'a':'z')
+        SimpleColor(Symbol(rgb))
+    else
+        nothing
+    end
+end
+
+"""
+    parse(::Type{SimpleColor}, rgb::String)
+
+An analogue of `tryparse(SimpleColor, rgb::String)` (which see),
+that raises an error instead of returning `nothing`.
+"""
+function parse(::Type{SimpleColor}, rgb::String)
+    color = tryparse(SimpleColor, rgb)
+    !isnothing(color) ||
+        throw(ArgumentError("invalid color \"$rgb\""))
+    color
+end
+
+"""
+A [`Face`](@ref) is a collection of graphical attributes for displaying text.
+Faces control how text is displayed in the terminal, and possibly other
+places too.
+
+Most of the time, a [`Face`](@ref) will be stored in the [`FACES`](@ref) dict as
+a unique association with a *face name* Symbol, and will be most often referred
+to by this name instead of the `Face` object itself.
+
+## Attributes
+
+All attributes can be set via the keyword constructor, and default to `nothing`.
+
+- `foreground` (a `SimpleColor`): The text foreground color.
+- `background` (a `SimpleColor`): The text background color.
+- `weight` (a `Symbol`): One of the symbols (from faintest to densest)
+  `:thin`, `:extralight`, `:light`, `:semilight`, `:normal`,
+  `:medium`, `:semibold`, `:bold`, `:extrabold`, or `:black`.
+  In terminals any weight greater than `:normal` is displayed as bold,
+  and in terminals that support variable-brightness text, any weight
+  less than `:normal` is displayed as faint.
+- `slant` (a `Symbol`): One of the symbols `:italic`, `:oblique`, or `:normal`.
+- `underline`, the text underline, which takes one of the following forms:
+  - a `Bool`: Whether the text should be underlined or not.\\
+  - a `SimpleColor`: The text should be underlined with this color.\\
+  - a `Tuple{Nothing, Symbol}`: The text should be underlined using the style
+    set by the Symbol, one of `:straight`, `:double`, `:curly`, `:dotted`,
+    or `:dashed`.\\
+  - a `Tuple{SimpleColor, Symbol}`: The text should be underlined in the specified
+    SimpleColor, and using the style specified by the Symbol, as before.
+- `strikethrough` (a `Bool`): Whether the text should be struck through.
+- `inverse` (a `Bool`): Whether the foreground and background colors should be
+  inverted.
+- `inherit` (a `Vector{Symbol}`): Names of faces to inherit from,
+  with earlier faces taking priority. All faces inherit from the `:default` face.
+"""
+struct Face
+    font::Union{Nothing, String}
+    foreground::Union{Nothing, SimpleColor}
+    background::Union{Nothing, SimpleColor}
+    weight::Union{Nothing, Symbol}
+    slant::Union{Nothing, Symbol}
+    underline::Union{Nothing, Bool, SimpleColor,
+                     Tuple{<:Union{Nothing, SimpleColor}, Symbol}}
+    strikethrough::Union{Nothing, Bool}
+    inverse::Union{Nothing, Bool}
+    inherit::Vector{Symbol}
+end
```

Comment:
There are some terminals that support "hidden" or "conceal" attributes. Do you know if that can be supported? There's also ways blink text in a terminal and I don't see that in the PR.

Reply:
Here's the attribute table from `Crossterm` for reference:

<img width="830" alt="image" src="https://github.com/JuliaLang/julia/assets/1813121/fbfa23b9-1f60-4f10-af5e-10202d13151c">

Reply:
From what I can tell, blinking and hidden text are rarely supported and (to me at least) not as clearly useful as the attributes currently supported.